cluster-autoscaler/cloudprovider/azure/azure_vms_pool.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"fmt"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/config"
+	"k8s.io/autoscaler/cluster-autoscaler/config/dynamic"
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+// VMsPool is single instance VM pool
+// this is a placeholder for now, no real implementation
+type VMsPool struct {
+	azureRef
+	manager       *AzureManager
+	resourceGroup string
+
+	minSize int
+	maxSize int
+
+	curSize int64
+	// sizeMutex       sync.Mutex
+	// lastSizeRefresh time.Time
+}
+
+// NewVMsPool creates a new VMsPool
+func NewVMsPool(spec *dynamic.NodeGroupSpec, am *AzureManager) *VMsPool {
+	nodepool := &VMsPool{
+		azureRef: azureRef{
+			Name: spec.Name,
+		},
+
+		manager:       am,
+		resourceGroup: am.config.ResourceGroup,
+
+		curSize: -1,
+		minSize: spec.MinSize,
+		maxSize: spec.MaxSize,
+	}
+
+	return nodepool
+}
+
+// MinSize returns the minimum size the cluster is allowed to scaled down
+// to as provided by the node spec in --node parameter.
+func (agentPool *VMsPool) MinSize() int {
+	return agentPool.minSize
+}
+
+// Exist is always true since we are initialized with an existing agentpool
+func (agentPool *VMsPool) Exist() bool {
+	return true
+}
+
+// Create creates the node group on the cloud provider side.
+func (agentPool *VMsPool) Create() (cloudprovider.NodeGroup, error) {
+	return nil, cloudprovider.ErrAlreadyExist
+}
+
+// Delete deletes the node group on the cloud provider side.
+func (agentPool *VMsPool) Delete() error {
+	return cloudprovider.ErrNotImplemented
+}
+
+// Autoprovisioned is always false since we are initialized with an existing agentpool
+func (agentPool *VMsPool) Autoprovisioned() bool {
+	return false
+}
+
+// GetOptions returns NodeGroupAutoscalingOptions that should be used for this particular
+// NodeGroup. Returning a nil will result in using default options.
+func (agentPool *VMsPool) GetOptions(defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error) {
+	// TODO(wenxuan): Implement this method
+	return nil, cloudprovider.ErrNotImplemented
+}
```

Comment:
https://github.com/kubernetes/autoscaler/pull/6677

Reply:
Yes I'm aware. This whole file is now just a place holder without any implementation. I will add the required methods in the following PRs

Reply:
Same concern. Thank you.